src/genie/libs/parser/ironware/show_optic.py

```diff
@@ -0,0 +1,153 @@
+    1/2   34.4335 C  -006.3507 dBm -006.7243 dBm   14.650 mA
+        Normal       Normal           Normal      Normal
+    1/3   38.9765 C  -006.9572 dBm -007.6980 dBm   20.800 mA
+        Normal       Normal           Normal      Normal
+    1/4   38.9492 C  -006.2324 dBm -007.1760 dBm   27.468 mA
+        Normal       Normal           Normal      Normal
+    1/5   50.5156 C  -005.8720 dBm -008.0106 dBm   41.800 mA
+        Normal       Normal           Normal      Normal
+    1/6   43.1718 C  -006.5403 dBm -004.4696 dBm   23.100 mA
+        Normal       Normal        High-Warn      Normal
+    """
+
+    def cli(self, slot, output=None):
+        if output is None:
+            out = self.device.execute(self.cli_command.format(slot=slot))
+        else:
+            out = output
+
+        result_dict = {}
+
+        p0 = re.compile(r'(^(?P<port>\d+\/\d+)\s+'
+                        r'(?P<temp>\d+.\d+|N\/A)(\sC|)\s+'
+                        r'(?P<tx>[-]?\d+.\d+|N\/A|NONE)(\s+dBm|'
+                        r'\s+dBm\/\s*\d+\s+uW|)\s+'
+                        r'(?P<rx>[-]?\d+.\d+|N\/A|NONE)(\s+dBm|'
+                        r'\s+dBm\/\s*\d+\s+uW|)\s+'
+                        r'(?P<tbc>[-]?\d+.\d+|N\/A)(\s+mA|))')
+
+        p1 = re.compile(r'(^(?P<temp>Normal|Low-Alarm|Low-Warn|'
+                        r'High-Warn|High-Alarm)\s+'
+                        r'(?P<tx>Normal|Low-Alarm|Low-Warn|'
+                        r'High-Warn|High-Alarm)\s+'
+                        r'(?P<rx>Normal|Low-Alarm|Low-Warn|'
+                        r'High-Warn|High-Alarm)\s+'
+                        r'(?P<tbc>Normal|Low-Alarm|Low-Warn|'
+                        r'High-Warn|High-Alarm))')
+
+        for line in out.splitlines():
+            line = line.strip()
+
+            m = p0.match(line)
+            if m:
+                if 'interfaces' not in result_dict:
+                    interfaces_dict = result_dict.setdefault('interfaces', {})
+
+                port = 'ethernet{0}'.format(m.groupdict()['port'])
+
+                temp = m.groupdict()['temp']
+                tx = m.groupdict()['tx']
+                rx = m.groupdict()['rx']
+                tbc = m.groupdict()['tbc']
+
+                int_dict = interfaces_dict[port] = {
+                    'temperature': {
+                        'value': self.can_float(temp)
+                    },
+                    'tx': {
+                        'value': self.can_float(tx)
+                    },
+                    'rx': {
+                        'value': self.can_float(rx)
+                    },
+                    'tx_bias_current': {
+                        'value': self.can_float(tbc)
+                    }
+                }
+                continue
+
+            m = p1.match(line)
+            if m:
+                temp = m.groupdict()['temp']
+                rx = m.groupdict()['rx']
+                tx = m.groupdict()['tx']
+                tbc = m.groupdict()['tbc']
+
```

Comment:
Just a note, you can do this as well and update the dict at once. I think this is trimmer personally, and I suggest that you apply this way, but I leave it to you to decide, you can try it and if you find it appropriate, apply it here and elsewhere
```
interfaces_dict.update({port: {
                                         'temperature': {'alarm': temp},
                                          'rx': {'alarm': rx},
                                          'tx': {'alarm': tx},
                                          'tx_bias_current': {'alarm': tbc},

}
})

Reply:
@KamyarZiabari It is my understanding that using `interfaces_dict.update` will actually update the inner keys for `temperature`, `rx`, `tx` and `tx_bias_current` with purely the subdict containing `{'alarm': xxx}`. E.g:

```python
>>> mydict1 = {
...     'interfaces': {
...             'temp': {
...                     'value': 1234
...             }
...     }
... }
>>>
>>> print(mydict1)
{'interfaces': {'temp': {'value': 1234}}}
>>>
>>>
>>> mydict1.update({'interfaces': {
...     'temp': {'alarm': True}
...     }
... }
... )
>>> print(mydict1)
{'interfaces': {'temp': {'alarm': True}}}
>>>
```